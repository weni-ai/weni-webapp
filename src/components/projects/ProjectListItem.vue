<template>
  <div class="weni-project-list-item">
    <div class="weni-project-list-item__header">
      <div class="weni-project-list-item__header__title">
        <span
          class="weni-project-list-item__header__name"
          :title="name.length > 25 ? name : null"
        >
          {{ name }}
        </span>
        <unnnic-tool-tip
          style="display: none"
          class="weni-project-list-item__header__time__wrapper"
          :text="$t('projects.created_at_tooltip')"
          side="left"
          :enabled="true"
        >
          <span class="weni-project-list-item__header__time">
            <unnnic-icon-svg size="xs" icon="time-clock-circle-1" />
            {{ time }}
          </span>
        </unnnic-tool-tip>
      </div>
      <div class="weni-project-list-item__header__buttons">
        <unnnic-tag
          @click.native="
            onClick({ name: 'home', params: { projectUuid: uuid } })
          "
          clickable
          :text="$t('projects.join')"
          scheme="aux-blue"
        />
        <unnnic-dropdown :open="false" position="bottom-left">
          <unnnic-icon-svg
            slot="trigger"
            size="sm"
            icon="navigation-menu-vertical-1"
            clickable
            scheme="neutral-cleanest"
          />
          <unnnic-dropdown-item
            @click="onClick({ name: 'project', params: { projectUuid: uuid } })"
          >
            <unnnic-icon-svg size="sm" icon="cog-1" />
            {{ $t('projects.config') }}
          </unnnic-dropdown-item>
        </unnnic-dropdown>
      </div>
    </div>
    <div class="weni-project-list-item__separator" />
    <div class="weni-project-list-item__status__list">
      <div
        class="weni-project-list-item__status"
        v-for="(status, index) in statusList"
        :key="index"
      >
        {{ status.title }}

        <div class="content">
          <div :class="['box', status.scheme]">
            <unnnic-icon-svg
              size="sm"
              :scheme="status.scheme"
              :icon="status.icon"
            />
          </div>

          <span class="weni-project-list-item__status__number">
            {{ status.count }}
          </span>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ProjectListItem',
  props: {
    uuid: {
      type: String,
      default: null,
    },
    name: {
      type: String,
      default: null,
    },
    time: {
      type: String,
      default: null,
    },
    owner: {
      type: String,
      default: null,
    },
    aiCount: {
      type: Number,
    },
    contactCount: {
      type: Number,
    },
    flowsCount: {
      type: Number,
    },
  },
  computed: {
    statusList() {
      return [
        {
          title: this.$t('projects.ai'),
          icon: 'science-fiction-robot-2',
          scheme: 'aux-blue',
          count: this.aiCount,
        },
        {
          title: this.$t('projects.flows'),
          icon: 'hierarchy-3-2',
          scheme: 'aux-purple',
          count: this.flowsCount,
        },
        {
          title: this.$t('projects.contacts'),
          icon: 'single-neutral-actions-1',
          scheme: 'aux-lemon',
          count: this.contactCount,
        },
      ];
    },
  },

  methods: {
    onClick(route) {
      this.$emit('click', route);
    },
    onClickBtn() {
      console.log('teste');
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@weni/unnnic-system/src/assets/scss/unnnic.scss';
.weni-project-list-item {
  padding: $unnnic-inset-sm;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
  border-radius: $unnnic-border-radius-md;
  font-family: $unnnic-font-family-secondary;
  min-height: 9.125rem;

  &__separator {
    margin: $unnnic-spacing-stack-sm 0;
    border: $unnnic-border-width-thinner solid $unnnic-color-neutral-lightest;
  }

  &--highlighted {
    color: $unnnic-color-brand-weni;
    font-weight: $unnnic-font-weight-bold;
  }

  &__header {
    display: flex;
    justify-content: space-between;
    align-items: center;

    color: $unnnic-color-neutral-cloudy;
    font-size: $unnnic-font-size-body-md;

    &__buttons {
      display: flex;
      align-items: center;

      .unnnic-icon {
        margin-left: $unnnic-spacing-inline-xs;
      }

      .unnnic-dropdown .unnnic-dropdown__content > a {
        display: flex;
        align-items: center;
        justify-content: space-between;

        > .unnnic-icon {
          margin-right: $unnnic-spacing-inline-xs;
          margin-left: 0;
        }
      }
    }

    &__title {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      width: calc(100% - 80px);
      margin-right: $unnnic-spacing-inline-xs;
    }

    &__time {
      text-align: right;
      flex: 1;

      &__wrapper {
        display: inline-flex;
        align-items: center;
      }
    }
    &__name {
      font-weight: $unnnic-font-weight-bold;
      color: $unnnic-color-neutral-darkest;
      font-size: $unnnic-font-size-body-lg;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      max-width: 100%;
    }
  }

  &__status {
    color: $unnnic-color-neutral-dark;
    font-size: $unnnic-font-size-body-gt;
    flex: 1;

    .box {
      border-radius: $unnnic-border-radius-sm;
      padding: $unnnic-spacing-inset-nano / 2;
      display: inline-block;
      margin-right: $unnnic-spacing-inline-xs;

      &.aux-blue {
        background-color: rgba(
          $unnnic-color-aux-blue,
          $unnnic-opacity-level-extra-light
        );
      }

      &.aux-purple {
        background-color: rgba(
          $unnnic-color-aux-purple,
          $unnnic-opacity-level-extra-light
        );
      }

      &.aux-lemon {
        background-color: rgba(
          $unnnic-color-aux-lemon,
          $unnnic-opacity-level-extra-light
        );
      }
    }

    .content {
      margin: $unnnic-spacing-stack-xs 0 0 0;
    }

    &__number {
      color: $unnnic-color-neutral-darkest;
      font-weight: $unnnic-font-weight-black;
    }

    &__list {
      display: flex;
      justify-content: space-between;
    }
  }
}
</style>
