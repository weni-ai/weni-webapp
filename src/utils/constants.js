export const PROJECT_COMMERCE = 2;
export const PROJECT_GENERAL = 1;