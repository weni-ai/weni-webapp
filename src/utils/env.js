import YAML from 'yaml';

export default function getEnv(name) {
  const value =
    process.env?.[name] || window?.configs?.[name] || import.meta.env[name];

  if (/(^|_)yaml($|_)/i.test(name)) {
    return YAML.parse(value);
  }

  return value;
}
