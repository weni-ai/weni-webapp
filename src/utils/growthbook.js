import { reactive } from 'vue';
import { configureCache, GrowthBook } from '@growthbook/growthbook';
import globalStore from '@/store';
import getEnv from '@/utils/env';

const gbKey = Symbol('growthbook');

const createGrowthBookInstance = () => {
  return reactive(
    new GrowthBook({
      apiHost: getEnv('GROWTHBOOK_API_HOST'),
      clientKey: getEnv('GROWTHBOOK_CLIENT_KEY'),
      attributes: {
        email: globalStore?.state?.Account?.profile?.email || '',
        weni_project: globalStore?.state?.Project?.currentProject?.uuid || '',
        weni_org: globalStore?.state?.Org?.currentOrg?.uuid || '',
      },
    })
  );
};

configureCache({
  cacheKey: 'gbFeaturesCache',
  maxAge: 1000 * 60 * 60 * 0.5,
});

const gbInstance = createGrowthBookInstance();

const initializeGrowthBook = async () => {
  if (!getEnv('GROWTHBOOK_API_HOST') || !getEnv('GROWTHBOOK_CLIENT_KEY')) {
    console.warn('GrowthBook configuration missing');
    return null;
  }

  try {
    await gbInstance.init();
    return gbInstance;
  } catch (e) {
    console.error('Error initializing GrowthBook:', e);
    return null;
  }
};

const getGrowthBook = () => {
  return gbInstance;
};

export { gbKey, gbInstance, initializeGrowthBook, getGrowthBook };