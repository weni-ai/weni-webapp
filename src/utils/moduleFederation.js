import { defineAsyncComponent } from 'vue';
import * as Sentry from '@sentry/browser';

/**
 * Creates a safe async component with automatic defineAsyncComponent wrapper
 * @param {Function} importFn - The import function (e.g., () => import('remote/component'))
 * @param {any} fallback - Fallback component if import fails
 * @returns {Component} - Vue async component ready to use
 */
export function safeAsyncComponent(importFn) {
  return defineAsyncComponent(async () => {
    try {
      return await importFn();
    } catch (error) {
      console.error(
        '[Module Federation] Failed to load remote component:',
        error.message,
      );

      Sentry.captureException(error, {
        tags: {
          module_federation: true,
          error_type: 'component_load_failed',
        },
        extra: {
          errorMessage: error.message,
          stack: error.stack,
        },
      });
    }
  });
}

/**
 * Tries to import a module with retries
 * @param {Function} importer - The import function (e.g., () => import('remote/component'))
 * @param {string} path - The import path for logging purposes
 * @param {number} maxAttempts - The maximum number of attempts
 * @param {number} delay - The delay between attempts
 * @returns {Promise<Object>} - The imported object or null
 */
export async function tryImportWithRetries(
  importer,
  path,
  maxAttempts = 3,
  delay = 2000,
) {
  for (let attempt = 1; attempt <= maxAttempts; attempt++) {
    const module = await safeImport(importer, path);
    if (module && JSON.stringify(module) !== '{}') return module;

    console.warn(`Import attempt ${attempt} failed for ${path}, retrying...`);

    if (attempt < maxAttempts) {
      await new Promise((resolve) => setTimeout(resolve, delay));
    } else {
      break;
    }
  }

  const errorMessage = `Failed to load ${path} after ${maxAttempts} attempts`;

  console.error(errorMessage);

  Sentry.captureException(new Error(errorMessage), {
    tags: { module_federation: true, import_path: path },
  });

  return null;
}

/**
 * Safe import for files
 * @param {Function} importFn - The import function (e.g., () => import('remote/locales/en'))
 * @param {string} importPath - The import path for logging purposes
 * @returns {Promise<Object>} - The imported object or empty object
 */
export async function safeImport(importFn, importPath) {
  try {
    const module = await importFn();
    return module.default || module;
  } catch (error) {
    console.error(
      `[Module Federation] ${importPath} unavailable:`,
      error.message,
    );

    Sentry.captureException(error, {
      tags: { module_federation: true, import_path: importPath },
    });

    return null;
  }
}
