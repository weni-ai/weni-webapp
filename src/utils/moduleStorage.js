/**
 * Utility for managing localStorage with module-specific prefixes
 * to avoid conflicts between Module Federation modules
 */

const MODULE_PREFIX = 'connect_';

class ModuleStorage {
  constructor(prefix = MODULE_PREFIX) {
    this.prefix = prefix;
  }

  /**
   * Get prefixed key
   * @param {string} key - Original key
   * @returns {string} Prefixed key
   */
  _getPrefixedKey(key) {
    return `${this.prefix}${key}`;
  }

  /**
   * Set item in localStorage with module prefix
   * @param {string} key - Storage key
   * @param {any} value - Value to store
   */
  setItem(key, value, options = { useSession: false }) {
    const prefixedKey = this._getPrefixedKey(key);
    const serializedValue =
      typeof value === 'string' ? value : JSON.stringify(value);

    if (options.useSession) {
      window.sessionStorage.setItem(prefixedKey, serializedValue);
    } else {
      window.localStorage.setItem(prefixedKey, serializedValue);
    }
  }

  /**
   * Get item from localStorage with module prefix
   * @param {string} key - Storage key
   * @param {any} defaultValue - Default value if key doesn't exist
   * @returns {any} Retrieved value
   */
  getItem(key, defaultValue = null, options = { useSession: false }) {
    const prefixedKey = this._getPrefixedKey(key);
    const value = options.useSession
      ? window.sessionStorage.getItem(prefixedKey)
      : window.localStorage.getItem(prefixedKey);

    if (value === null) {
      return defaultValue;
    }

    try {
      return JSON.parse(value);
    } catch {
      return value;
    }
  }

  /**
   * Remove item from localStorage with module prefix
   * @param {string} key - Storage key
   */
  removeItem(key, options = { useSession: false }) {
    const prefixedKey = this._getPrefixedKey(key);
    if (options.useSession) {
      window.sessionStorage.removeItem(prefixedKey);
    } else {
      window.localStorage.removeItem(prefixedKey);
    }
  }

  /**
   * Clear all items with this module's prefix
   */
  clear(options = { useSession: false }) {
    const keysToRemove = [];
    for (let i = 0; i < window.localStorage.length; i++) {
      const key = options.useSession
        ? window.sessionStorage.key(i)
        : window.localStorage.key(i);
      if (key && key.startsWith(this.prefix)) {
        keysToRemove.push(key);
      }
    }
    keysToRemove.forEach((key) => {
      if (options.useSession) {
        window.sessionStorage.removeItem(key);
      } else {
        window.localStorage.removeItem(key);
      }
    });
  }
}

export const moduleStorage = new ModuleStorage();

export { ModuleStorage };