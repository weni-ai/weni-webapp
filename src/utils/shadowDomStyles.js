/**
 * Utility to manage styles in Shadow DOM
 * Observes and clones styles from a specific module to the Shadow Root
 */

/**
 * Creates a style observer for a Shadow DOM
 * @param {string} moduleName - Module name (for logs)
 * @returns {Object} - Object with start and stop methods
 */
export function createShadowStyleObserver(moduleName = 'Module') {
  let styleObserver = null;
  let observedStyles = new WeakSet();

  /**
   * Clones a style node (<link> or <style>) to the Shadow Root
   * IMPORTANT: Removes from main DOM to prevent style leakage
   *
   * @param {HTMLElement} node - Style node to be cloned
   * @param {ShadowRoot} shadowRoot - Target Shadow Root
   */
  function cloneStyleToShadowRoot(node, shadowRoot) {
    // Mark as observed to avoid duplicates
    observedStyles.add(node);

    if (node.tagName === 'LINK') {
      // For <link>, create a copy
      const link = document.createElement('link');
      link.rel = 'stylesheet';
      link.href = node.href;

      // Wait for loading before removing from main DOM
      link.onload = () => {
        console.log(
          `‚úÖ [Shadow Root ${moduleName}] <link> loaded, removing from <head>`,
        );
        // Remove from main DOM after loading in Shadow Root
        if (node.parentNode) {
          node.remove();
        }
      };

      link.onerror = () => {
        console.warn(
          `‚ö†Ô∏è [Shadow Root ${moduleName}] Error loading <link>, keeping in <head>`,
        );
      };

      shadowRoot.appendChild(link);
      console.log(`üé® [Shadow Root ${moduleName}] <link> cloned`);
    } else if (node.tagName === 'STYLE') {
      // For <style>, clone the content
      const style = document.createElement('style');
      style.textContent = node.textContent;
      shadowRoot.appendChild(style);
      console.log(
        `üé® [Shadow Root ${moduleName}] <style> cloned to Shadow Root`,
      );

      // Remove from main DOM immediately (already in Shadow Root)
      if (node.parentNode) {
        node.remove();
        console.log(
          `‚úÖ [Shadow Root ${moduleName}] <style> removed from <head>`,
        );
      }
    }
  }

  /**
   * Copies existing module styles that are already in <head>
   *
   * @param {ShadowRoot} shadowRoot - Target Shadow Root
   * @param {string} moduleUrl - Module URL to filter styles
   */
  function copyExistingStyles(shadowRoot, moduleUrl) {
    const existingStyles = document.head.querySelectorAll(
      'style, link[rel="stylesheet"]',
    );

    existingStyles.forEach((node) => {
      const isModuleStyle =
        (node.href && node.href.includes(moduleUrl)) ||
        (node.tagName === 'STYLE' && node.textContent);

      if (isModuleStyle && !observedStyles.has(node)) {
        console.log(
          `üé® [Shadow Root ${moduleName}] Copying existing style:`,
          node,
        );
        cloneStyleToShadowRoot(node, shadowRoot);
      }
    });
  }

  /**
   * Starts style observation
   *
   * @param {ShadowRoot} shadowRoot - Target Shadow Root
   * @param {string} moduleUrl - Module URL to filter styles
   */
  function start(shadowRoot, moduleUrl) {
    if (!shadowRoot) {
      console.error(`‚ùå [Shadow Root ${moduleName}] Shadow Root not provided`);
      return;
    }

    console.log(`üîç [Shadow Root ${moduleName}] Starting style observation`);

    // Copy existing module styles that are already in <head>
    copyExistingStyles(shadowRoot, moduleUrl);

    // Create observer for new styles
    styleObserver = new MutationObserver((mutations) => {
      mutations.forEach((mutation) => {
        mutation.addedNodes.forEach((node) => {
          if (node.nodeType === 1) {
            // Element node
            const isStyleNode =
              node.tagName === 'STYLE' ||
              (node.tagName === 'LINK' && node.rel === 'stylesheet');

            if (isStyleNode && !observedStyles.has(node)) {
              const isModuleStyle =
                (node.href && node.href.includes(moduleUrl)) ||
                (node.tagName === 'STYLE' && node.textContent);

              if (isModuleStyle) {
                console.log(
                  `üé® [Shadow Root ${moduleName}] New style detected:`,
                  node,
                );
                cloneStyleToShadowRoot(node, shadowRoot);
              }
            }
          }
        });
      });
    });

    // Observe changes in <head>
    styleObserver.observe(document.head, {
      childList: true,
      subtree: false,
    });

    console.log(`‚úÖ [Shadow Root ${moduleName}] Style observation enabled`);
  }

  /**
   * Stops style observation
   */
  function stop() {
    if (styleObserver) {
      styleObserver.disconnect();
      styleObserver = null;
      // WeakSet doesn't have clear() method, so we create a new one
      observedStyles = new WeakSet();
      console.log(`‚úÖ [Shadow Root ${moduleName}] Style observation disabled`);
    }
  }

  return {
    start,
    stop,
  };
}
