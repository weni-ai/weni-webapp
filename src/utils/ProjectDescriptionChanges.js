import store from '../store';
import { moduleStorage } from './moduleStorage';

export default {
  register({ projectUuid, description }) {
    let projectDescriptions = JSON.parse(
      moduleStorage.getItem('project-descriptions') || '[]',
    );

    const projectDescription = projectDescriptions.find(
      ({ uuid }) => uuid === projectUuid,
    );

    if (projectDescription) {
      projectDescription.description = description;
      projectDescription.updatedAt = new Date();
    } else {
      projectDescriptions.push({
        uuid: projectUuid,
        description,
        updatedAt: new Date(),
      });
    }

    moduleStorage.setItem(
      'project-descriptions',
      JSON.stringify(projectDescriptions),
    );
  },

  project({ projectUuid }) {
    if (store.state.Project.currentProject?.uuid === projectUuid) {
      return store.state.Project.currentProject;
    }

    return store.state.Project.projects
      .map(({ data }) => data)
      .flat()
      .find(({ uuid }) => uuid === projectUuid);
  },

  isChanged({ projectUuid }) {
    let changed = false;

    let projectDescriptions = JSON.parse(
      moduleStorage.getItem('project-descriptions') || '[]',
    );

    const projectDescription = projectDescriptions.find(
      ({ uuid }) => uuid === projectUuid,
    );

    const project = this.project({ projectUuid });

    if (
      projectDescription &&
      project &&
      projectDescription.description !== project.description
    ) {
      project.description = projectDescription.description;
      changed = true;
    }

    const remove = projectDescriptions.filter(
      (projectDescription) =>
        (new Date() - new Date(projectDescription.updatedAt)) / 1000 > 6,
    );

    if (remove.length) {
      const final = projectDescriptions.filter(
        ({ uuid }) => !remove.some((toRemove) => toRemove.uuid === uuid),
      );

      if (final.length) {
        moduleStorage.setItem('project-descriptions', JSON.stringify(final));
      } else {
        moduleStorage.removeItem('project-descriptions');
      }
    }

    return changed;
  },
};
