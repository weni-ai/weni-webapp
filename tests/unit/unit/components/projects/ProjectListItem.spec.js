import { vi } from 'vitest';
import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';

import ProjectListItem from '@/components/projects/ProjectListItem.vue';
import { PROJECT_COMMERCE } from '@/utils/constants';

import { project, authorizations } from '../../../__mocks__/index';

vi.mock('@/store/featureFlags', () => ({
  useFeatureFlagsStore: () => ({
    flags: {
      newConnectPlataform: false,
      agentsTeam: false
    },
    instance: {
      context: {}
    },
    isWeniProjectOn: vi.fn().mockReturnValue(false)
  })
}));

vi.mock('@/api/brain', () => ({
  customization: {
    get: vi.fn().mockResolvedValue({ data: { team: { human_support: true } } })
  }
}));

describe('ProjectListItem.vue', () => {
  let getters;
  let actions;
  let store;
  let wrapper;

  beforeEach(() => {
    getters = {
      currentOrg: () => {
        return {
          name: 'oi',
          uuid: '12',
          inteligence_organization: 'topp',
          authorization: 'true',
        };
      },
    };
    actions = {
      setCurrentProject: vi.fn(),
      createProject: vi.fn(),
      createOrUpdateProjectAuthorization: vi.fn(),
    };

    store = createStore({
      getters,
      actions,
      modules: {
        Account: {
          state: {
            profile: {
              username: 'test',
            },
          },
        },
      },
    });

    wrapper = shallowMount(ProjectListItem, {
      global: {
        plugins: [store],
        stubs: {
          RouterLink: RouterLinkStub,
          UnnnicIconSvg: true,
          ProjectListItem: true,
          InfiniteLoading: true,
          UnnnicCardProject: true,
          UnnnicButton: true,
          UnnnicMultiSelect: true,
          UnnnicDropdown: true,
          UnnnicInput: true,
          UnnnicDropdownItem: true,
          UnnnicToolTip: true,
          UnnnicTag: true,
        },
      },
      props: {
        project: {
          ...project,
          project_mode: PROJECT_COMMERCE
        },
        name: 'name',
        time: 'time',
        owner: 'pwmer',
        aiCount: 12,
        contactCount: 12,
        flowsCount: 12,
        authorizations,
        pendingAuthorizations: {
          count: 0,
          users: [],
        },
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('test onclick', async () => {
    await wrapper.vm.onClick('route');
    expect(wrapper.emitted().click).toBeTruthy();
  });

  it('test computed statuList', async () => {
    await wrapper.setData({
      aiCount: 12,
      flowsCount: 12,
      contactCount: 12,
    });

    expect(wrapper.vm.statusList).toEqual([
      {
        title: 'Intelligences',
        icon: 'neurology',
        scheme: 'aux-blue',
        count: 12,
      },
      {
        title: 'Flows',
        icon: 'account_tree',
        scheme: 'aux-purple',
        count: 12,
      },
      {
        title: 'Contacts',
        icon: 'person',
        scheme: 'aux-lemon',
        count: 12,
      },
    ]);
  });

  it('test if you can change members', () => {
    expect(wrapper.vm.canManageMembers).toBeTruthy();
  });

  describe('isEnabledSettings', () => {
    it('should return true when neither feature is enabled', async () => {
      await wrapper.setData({
        isOrgEnabledAgentBuilder2: false,
        isProjectEnabledAgentBuilder2: false
      });

      expect(wrapper.vm.isEnabledSettings).toBe(true);
    });

    it('should return false when all disabling conditions are met', async () => {
      await wrapper.setData({
        isOrgEnabledAgentBuilder2: true,
        isProjectEnabledAgentBuilder2: true,
        isEnabledUserNewPlatform: true,
        isAgentBuilder2: true,
        isHumanServiceEnabled: false
      });

      expect(wrapper.vm.isEnabledSettings).toBe(false);
    });

    it('should return true when human service is enabled', async () => {
      await wrapper.setData({
        isOrgEnabledAgentBuilder2: true,
        isProjectEnabledAgentBuilder2: true,
        isEnabledUserNewPlatform: true,
        isAgentBuilder2: true,
        isHumanServiceEnabled: true
      });

      expect(wrapper.vm.isEnabledSettings).toBe(true);
    });
  });
});
