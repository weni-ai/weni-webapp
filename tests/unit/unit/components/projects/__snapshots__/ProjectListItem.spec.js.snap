// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectListItem.vue > renders a snapshot 1`] = `
<div
  class="weni-project-list-item"
  data-v-77d2ab83=""
  owner="pwmer"
>
  <unnnic-card-project-stub
    canupdatestatus="true"
    clickable=""
    data-v-77d2ab83=""
    name="name"
  />
</div>
`;
